Formula/c/cpuinfo.rb

```diff
@@ -0,0 +1,46 @@
+class Cpuinfo < Formula
+  desc "CPU INFOrmation library and associated command-line tools"
+  homepage "https://github.com/pytorch/cpuinfo"
+  url "https://github.com/pytorch/cpuinfo/archive/3c8b1533ac03dd6531ab6e7b9245d488f13a82a5.tar.gz"
```

Comment:
we need a tagged release.

Reply:
They don't exist :man_shrugging:. Debian simply names the package versions `gitYYYYMMDD.${GITHASH}`. https://packages.debian.org/search?keywords=cpuinfo

Reply:
For what it's worth, @SMillerDev and @chenrui333, I see several other formulae where the "releases" are simply date-plus-Git-hash slugs.


> the only catch is we need a tagged release. (as we only ship stable releases at best efforts)

Is this requirement a recently-added one?

Reply:
No, it's not a new requirement. `luajit` is a one-off exception, but there should not be any others. I'll do a review when I have time to confirm this.

Reply:
@p-linnane, if I simply `git grep -P -C5 'version "20\d\d' Formula/`, I find **numerous** other formulae in which the version identifiers used by Homebrew are simply dates and/or Git hashes, with no sign of a clear versioning or tagging strategy that originates in the upstream projects.

Here is an example that appears nearly identical to this one:
https://github.com/Homebrew/homebrew-core/blob/262db065cc1ce35f13b143a145b7580e9447a00f/Formula/j/jsmin.rb#L4-L11

Reply:
I just ran that, and looked through the results. `jsmin` is the only one that matches what you are talking about here. All other results are us having to clarify the string we want to use as the version, because it can't be parsed from the filename. Additionally, it appears `jsmin` is the way it is because the repo has no tags, but the website states a release date, so that's what's being used.

Reply:
> jsmin is the only one that matches what you are talking about here. All other results are us having to clarify the string we want to use as the version, because it can't be parsed from the filename.

How about, say, https://github.com/Homebrew/homebrew-core/blob/master/Formula/i/ipbt.rb#L3-L7?

The "release" is simply a tarball with a date and a Git hash embedded in it. The homepage simply mentions it as:
>A Unix source archive of IPBT is available here:
>
>[ipbt-20240501.bc876ea.tar.gz](https://www.chiark.greenend.org.uk/~sgtatham/ipbt/ipbt-20240501.bc876ea.tar.gz)

… without otherwise describing any strategy or approach to versioning.

I guess what I'd like to clarify here is: what is a minimum threshold for what counts as "versions that come from upstream"?

Reply:
The regex you want is probably something more like
```
git grep -P -C5 'url "https://github.com/[^/]*/[^/]*/archive/[0-9a-f]*\.tar\.gz"' -- Formula/
```
This produces a number of results, but most of the matches belong to `resource`s (which don't need version tags). There are three formulae `url`s that match (i.e. the match doesn't belong to a `resource`): `luajit`, `jsmin` and `solid`.

Of those, `solid` and `luajit` used to provide tags/versioned download URLs (see the repo Git log for details). `jsmin` was likely added before we adopted the policy[^1] of only packaging software where upstream has version tags.

If you really want this in Homebrew/core, ask upstream if they're willing to tag their releases. Otherwise, you can host this formula in your own tap.

[^1]: https://docs.brew.sh/Acceptable-Formulae#stable-versions